src/Sentry/ISpan.cs

```diff
@@ -76,4 +76,20 @@ span switch
             SpanTracer tracer => tracer.Transaction,
             _ => throw new ArgumentOutOfRangeException(nameof(span), span, null)
         };
+
+    /// <summary>
+    /// Gets the parent span for database operations. This is the last active non-database span, which might be the
+    /// transaction root, or it might be some other child span of the transaction (such as a web request).
+    /// </summary>
+    /// <remarks>
+    /// Used by EF, EF Core, and SQLClient integrations.
+    /// </remarks>
+    internal static ISpan GetDbParentSpan(this ISpan span)
+    {
+        var transaction = span.GetTransaction();
+        return transaction.Spans
+                   .OrderByDescending(x => x.StartTimestamp)
```

Comment:
Is it possible in, global mode, this list can be mutated while we're iterating here?

Reply:
Perhaps, but if so then we already have that problem.

https://github.com/getsentry/sentry-dotnet/blob/de3cf7288aec646842f32d59ef88eec9d859fbf1/src/Sentry/TransactionTracer.cs#L298-L300

This one does the same thing, just with the additional filter to make sure we don't grab a connection span in the case of multiple connections on separate threads.